js/renovation/component_wrapper/utils/__tests__/update-props-immutable.test.ts

```diff
@@ -100,3 +100,12 @@ it('change object item in nested array option', () => {
   expect(props.nestedArray).not.toBe(nestedArray);
   expect(props.nestedArray[1]).toBe('item2 modified');
 });
+
+it('change object item in double nested array option', () => {
```

Comment:
Check more complex scenario { item: { objectProp: {}, items: [{ value: 'item1' }, { value: 'item2' }] } } and check that only data related to 'item1' mutated